### DISCUSSION TITLE
Sözdizim Tartışmaları

### DISCUSSION THREAD

**Discussion by volkantash** [turkce-yazilim-konati/yazilimca#53] [Ölçünler]
Selamun aleyküm, Türkçe tabanlı sözdizimler konusunda düşüncelerimizi bu konuda tartışıyoruz.

**Comment by volkantash** [comment]
İngilizce tabanlı yazılım dilinde eylem nesneden önce geldiği için şu biçimde bir düzen kurulmaktadır: 

print("")

Türkçe tabanlı yazılım dilinde ise Türkçeye göre düşündüğümüzde eylem nesneden sonra gelmelidir. Şu türde bir düzen kurulmalıdır.

("")bas

Şimdi bu türde düzen kurduğumuzda bir sıkıntı yaşamaktayız. Yazdığımız kod satırda uzadıkça eylem daha da sağda kalıyor. Bu nedenle sağ yanda bir çok ileride bir çok beride görüntü oluşuyor. Bunu aşmak için ise iki yöntem önerim var.

1. Yöntem
Kodları sağa dayalı yazmak.

```
                           a==10 ise:
             ("İçerisi dolu")bas
```

2. Yöntem
Nesneyi yazdıktan sonra alt satırda eylemi neyin yazmak.

```
a==10
ise:
   ("İçerisi dolu")
   bas
```

**Reply by turkmenistanli** [reply]
1 ve 2. yöntemleri tek bir yöntemdebirleştirip yöntemin 1. ve 2. aşamaları olarak çalışmamız mümkün müdür?

**Reply by volkantash** [reply]
Ne yazık ki öyle bir durum burada doğmuyor. Bu yazışlar daha çok Türkçenin özelliklerini ilk bakışta anlayabilmemize yaklaşmak amacıyla yazılan örnekler. 1. Yöntem sağa dayalı olarak yazmak iken, 2. Yöntem sola dayalı yazmak olup bir satır nesne bir satır buyruk olarak karşımıza çıkar. Bu tür arayışlar kodları daha anlaşılır ve kolay yazma arayışlarıdır. İngilizce tabanlı düzenlerden ötürü oluşturulan yazışlara alışık olanlar ilk başta bazı yöntemleri anlamayabilir.

**Comment by erhanbaris** [comment]
1. Yontem
Ne kadar sagda olacak?
Yeni bir kosul eklenince ondan onceki butun kosullari tekrar saga mi oteleyecegiz?

2. Yontem
Her satir kendi icinde sorumlugunu yerine getirir. Bir once ki yada bir sonra ki komut satirindan sorumlu olmasi karmasaya neden olur. 
Programlama dilleri yapisi geregi tutarli olmasi gerekir. Oldukca yalin, acik, yazmasi ve duzenlemesi kolay olmalidir.

**Reply by volkantash** [reply]
1. Yöntem
Yazılar sağa dayalı olacak. Sola dayalı yazıda ne oluyor ise, sağa dayalı yazıda da aynısı olmaktadır. Tarafımdan denenmiştir.

2. Yöntem
Şöyle yazarsak sizce sorun yaşar mıyız?
```
(a==1
)ise:
    ("İçerisi dolu"
    )bas
```

**Comment by volkantash** [comment]
Şöyle sağa dayalı bir sözdizim üzerinde alıştırma yapıyorum. Şimdilik çok iyi gidiyor.
![resim](https://user-images.githubusercontent.com/25375593/121827001-3b450480-ccc3-11eb-9fcd-bcba67117ce5.png)

**Reply by lemreaskin** [reply]
Bence güzel Türkçemiz gibi sola dayalı bir sözdizimi üzerinde çalışsak daha iyi olabilir. En azından görsel açıdan daha aşina olduğumuz bir düzen olur diye düşünüyorum.

**Reply by volkantash** [reply]
Öncelikle Türkçe sola dayalı dil değil, yazıda soldan sağa doğru okunan dil. Bilinen en eski biçimi sağdan sola doğru okunan biçim idi. Sola dayalı sözdizimi zaten var. Üstteki sözdizime alışık olmadığımız için ilk başta tuhaf algılayabiliriz. Zamanla oturur. Önemli olan Türkçe dil yapısına uygun bir yazış yöntemi ortaya koyabilmemiz. Çünkü yüklem gibi durumlar sonda oluyor. Bu durumdan yararlanıp sağa dayalı yazmayı deniyorum.

Sağa dayalı düzende alışkanlığı bir yana bırakırsak yapıca eksiklerimiz neler? Bir şeyler görebiliyor musunuz?

Bir de şimdi eklediğim görselde sola dayalı düzen var. Hâlâ birkaç durum İngilizce tabanlı sözdizim etkilerinden kalan alışkanlıklarla dolu. Bunlardan sıyrılarak irdelemeye çalışıyorum. Sola dayalı düzenler üzerinde de çalışıyorum. Sıkıntı yok, bunalma yok.
![resim](https://user-images.githubusercontent.com/25375593/121869456-bc73ba00-cd0a-11eb-9f49-332176689f63.png)

**Reply by erhanbaris** [reply]
Saga dayali ekran goruntusu fazla karmasik. Hersey ic ice girmis gibi. Okumasi ve anlamasi cok zor. Durduk yere icat cikartmaya gerek yok.
Onemli olan insanlarin kullanmak isteyecekleri birsey yapmak, tamamen farkli birsey yapip sadece teoride kalmasi bir fayda saglamiyor.

**Reply by lemreaskin** [reply]
Acaba **fonk** _(değer1,değer2)_ **_fonksiyonİsmi:_** şeklinde, sözdiziminde soldan sağa doğru bir tanımlama yapsak nasıl olur? Farklı bir açıdan bakmak istedim.

**Reply by lemreaskin** [reply]
> Saga dayali ekran goruntusu fazla karmasik. Hersey ic ice girmis gibi. Okumasi ve anlamasi cok zor. Durduk yere icat cikartmaya gerek yok.
> 
> Onemli olan insanlarin kullanmak isteyecekleri birsey yapmak, tamamen farkli birsey yapip sadece teoride kalmasi bir fayda saglamiyor.

Açıkçası katılıyorum, _Amerika'yı yeniden keşfetmeye_ gerek yok. Bence sözdizim içerisinde ufak farklılıklar yapmak daha iyi olabilir.

**Reply by volkantash** [reply]
Söylediğim üzere sağa dayalı düzenin tuhaf gelmesi doğal. Hatta keşfedilmemiş bir alan üzerinde çalışıyoruz diyebiliriz. Ana dilimiz Farsça falan olsa idi neredeyse bu denemelerin çoğunu yapmaz idik. Sola dayalı çalışmalardan bir örneği sizin için hızlıca yazdım. Önemli olan sola, sağa dayalı olması değil. Önemli olan Türkçe ile düşünürken kodların sözdizimi düşünceyi sağa sola sürtmemesi. Türkçe İngilizce gibi bir dil değil ki Amerikayı yeniden keşfetmeye gerek yok anlayışını güdelim. İngilizce tabanlı yapıdan sıyrılmaya çalışmalıyız. :)

![resim](https://user-images.githubusercontent.com/25375593/121877649-80912280-cd13-11eb-8c00-38d597e14876.png)

**Reply by volkantash** [reply]
> 
> 
> Acaba **fonk** _(değer1,değer2)_ **_fonksiyonİsmi:_** şeklinde, sözdiziminde soldan sağa doğru bir tanımlama yapsak nasıl olur? Farklı bir açıdan bakmak istedim.

Ben bu görüşteyim. Başka başlıklardaki geçmiş yorumlarımda örnekleri görebilirsiniz. ^.^

**Reply by lemreaskin** [reply]
> > 
> 
> > 
> 
> > Acaba **fonk** _(değer1,değer2)_ **_fonksiyonİsmi:_** şeklinde, sözdiziminde soldan sağa doğru bir tanımlama yapsak nasıl olur? Farklı bir açıdan bakmak istedim.
> 
> 
> 
> Ben bu görüşteyim. Başka başlıklardaki geçmiş yorumlarımda örnekleri görebilirsiniz. ^.^

Yani bence eğer yenilik yapacaksak, sözdizimi üzerinde böyle bir şey yapabiliriz. Tabii bu durum öğrenme zorluğu getirir mi onu da düşünmek gerekiyor. Eğer öğrenme zorluğu fazla olursa tercih eden sayısı az olacaktır. Çünkü hemen hemen her programlama dilinin benzer noktaları var. O nedenle "bir dili öğrenirsen diğer dilleri öğrenmen daha kolay olur" gibi bir anlayış hakimdir diye düşünüyorum.

Dediğim gibi bana kalırsa böyle bir kullanım kötü olmayabilir. Bununla birlikte, sözdizim yapısındaki bu farklılığın öğrenme zorluğuna etkisini de tartışmak gerekiyor.

**Reply by volkantash** [reply]
> 
> 
> > > 
> > 
> > 
> > > 
> > 
> > 
> > > Acaba **fonk** _(değer1,değer2)_ **_fonksiyonİsmi:_** şeklinde, sözdiziminde soldan sağa doğru bir tanımlama yapsak nasıl olur? Farklı bir açıdan bakmak istedim.
> > 
> > 
> > Ben bu görüşteyim. Başka başlıklardaki geçmiş yorumlarımda örnekleri görebilirsiniz. ^.^
> 
> Yani bence eğer yenilik yapacaksak, sözdizimi üzerinde böyle bir şey yapabiliriz. Tabii bu durum öğrenme zorluğu getirir mi onu da düşünmek gerekiyor. Eğer öğrenme zorluğu fazla olursa tercih eden sayısı az olacaktır. Çünkü hemen hemen her programlama dilinin benzer noktaları var. O nedenle "bir dili öğrenirsen diğer dilleri öğrenmen daha kolay olur" gibi bir anlayış hakimdir diye düşünüyorum.
> 
> Dediğim gibi bana kalırsa böyle bir kullanım kötü olmayabilir. Bununla birlikte, sözdizim yapısındaki bu farklılığın öğrenme zorluğuna etkisini de tartışmak gerekiyor.

Oluşturup denemeden, kendimizi denek kılmadan bilemeyiz. Şöyle işlevler de düşünüyorum.

sözcük(?)sözcük

Solunda olması ile sağında olmasının amaçlarını ortaya koymalıyız. 

Bir adım daha ileri gidip şuna dahi bakmalıyız.

sözcük.sözcük(?)sözcük.sözcük

Biraz müzik (https://soundcloud.com/fredjimusic/fredji-happy-life-original-mix)

**Reply by tolga-celik** [reply]
Bence ingilizce programlama dili için yapılmış kalıptaki sözcükleri türkçesiyle değiştirmek yerine türkçeye öxgü kalıplarla bir dil geliştirilmeli.

**Comment by volkantash** [comment]
Fonksiyonun Türkçe karşılığı işlev desek, yeni bir işlev oluşturma konusunda şöyle bir düzen güdülebilir.

```python

(a,b)toplama işlevi:
    a + b döndür
```

İşlevi (fonksiyonu) herhangi bir yerde kullanmak istediğimizde şöyle kullanabiliriz:
```python
(10,34)toplama
```
